packages/xo-web/src/xo-app/xostor/index.js

```diff
@@ -0,0 +1,559 @@
+  }),
+  provideState({
+    initialState: () => ({ onlyShowXostorPools: true }),
+    effects: {
+      toggleState,
+    },
+    computed: {
+      poolPredicate: (state, props) => pool => {
+        if (!state.onlyShowXostorPools) {
+          return true
+        }
+        const hostsPool = props.hostsByPoolId?.[pool.id]
+        return (
+          isGoodNumberOfHosts(hostsPool) && isXcpngPool(first(hostsPool)) && !alreadyHaveAXostor(props.srs[pool.id])
+        )
+      },
+      isPoolGoodNumberOfHosts: state => isGoodNumberOfHosts(state.hostsPool),
+      isPoolAlreadyHaveXostor: (state, props) => alreadyHaveAXostor(props.srs[state.poolId]),
+      isPoolCompatibleXostor: state =>
+        state.isXcpngPool && state.isPoolGoodNumberOfHosts && !state.isPoolAlreadyHaveXostor,
+    },
+  }),
+  injectState,
+  ({ effects, state }) => (
+    <Card>
+      <CardHeader>{_('pool')}</CardHeader>
+      <CardBlock>
+        <div>
+          <label>
+            <input
+              checked={state.onlyShowXostorPools}
+              name='onlyShowXostorPools'
+              onChange={effects.toggleState}
+              type='checkbox'
+            />{' '}
+            {_('onlyShowXostorRequirements', { type: _('pools') })}
+          </label>
+          <SelectPool onChange={effects.onPoolChange} predicate={state.poolPredicate} value={state.poolId} />
+          {state.hostsPool !== undefined && !state.isPoolCompatibleXostor && (
+            <div className='text-danger'>
+              {/* FIXME: add link of the documentation when ready */}
+              <a href='#' rel='noreferrer' target='_blank'>
+                {_('objectDoesNotMeetXostorRequirements', { object: <PoolRenderItem id={state.poolId} /> })}
+              </a>
+              <ul>
+                {!state.isXcpngPool && <li>{_('notXcpPool')}</li>}
+                {!state.isPoolGoodNumberOfHosts && <li>{_('wrongNumberOfHosts')}</li>}
+                {state.isPoolAlreadyHaveXostor && <li>{_('poolAlreadyHaveXostor')}</li>}
+              </ul>
+            </div>
+          )}
+          <em>
+            {/*
+              FIXME: If packages are already installed display: '"xcp-ng-release-linstor" and "xcp-ng-linstor" are already installed.'
+              Need to be done when backend is done.
+            */}
+            <Icon icon='info' /> {_('xostorPackagesWillBeInstalled')}
+          </em>
+        </div>
+      </CardBlock>
+    </Card>
+  ),
+])
+
+const NetworkCard = decorate([
+  provideState({
+    initialState: () => ({ onlyShowXostorNetworks: true }),
+    effects: {
+      toggleState,
+    },
+    computed: {
+      networksPredicate: (state, props) => network => {
+        const filterByPoolId = network.$pool === state.poolId
+        const pifs = network.PIFs
+        return state.onlyShowXostorNetworks
+          ? filterByPoolId && pifs.length > 0 && pifs.every(pifId => props.pifs[pifId].ip !== '')
+          : filterByPoolId
```

Comment:
Similarly, you can return a different predicate based on the value of `onlyShowXostorNetworks` instead of testing the condition for each network.

Reply:
But here, even though `onlyShowXostorNetworks === false`, we need to show only the networks of the selected pool.